classes/step1_form.php

```diff
@@ -0,0 +1,284 @@
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * File containing the step 1 form.
+ *
+ * @package     tool_pluginskel
+ * @subpackage  util
+ * @copyright   2016 Alexandru Elisei <alexandru.elisei@gmail.com>, David Mudrák <david@moodle.com>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once($CFG->libdir . '/formslib.php');
+
+/**
+ * Step 1 form.
+ *
+ * @package     tool_pluginskel
+ * @copyright   2016 Alexandru Elisei <alexandru.elisei@gmail.com>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class tool_pluginskel_step1_form extends moodleform {
+
+    /**
+     * The standard form definiton.
+     */
+    public function definition () {
+        $mform =& $this->_form;
+
+        $recipe = $this->_customdata['recipe'];
+        $component = $recipe['component'];
+        $templatevars = tool_pluginskel\local\util\manager::get_component_variables($component);
+        $features = tool_pluginskel\local\util\manager::get_features_variables();
+
+        $mform->addElement('header', 'pluginhdr', get_string('pluginhdr', 'tool_pluginskel'));
+        $mform->setExpanded('pluginhdr', true);
+
+        foreach ($templatevars as $variable) {
+
+            // The default element for a template variable is a text input.
+            if (empty($variable['hint']) || $variable['hint'] == 'text') {
+                $this->add_text($variable, $recipe);
+                continue;
+            }
+
+            // Template variables that are arrays will be added at the bottom of the page.
+            if ($variable['hint'] == 'multiple') {
+                continue;
+            }
+
+            if ($variable['hint'] == 'boolean') {
+                $this->add_advcheckbox($variable, $recipe, true);
+            }
+
+            if (!empty($variable['values'])) {
+                $this->add_select($variable, $recipe);
+            }
+        }
+
+        foreach ($features as $variable) {
+
+            // The default element for a template variable is a text input.
+            if (empty($variable['hint']) || $variable['hint'] == 'text') {
+                $this->add_text($variable, $recipe);
+                continue;
+            }
+
+            // Template variables that are arrays will be added at the bottom of the page.
+            if ($variable['hint'] == 'multiple') {
+                continue;
+            }
+
+            if ($variable['hint'] == 'boolean') {
+                $this->add_advcheckbox($variable, $recipe, true);
+            }
+
+            if (!empty($variable['values'])) {
```

Comment:
I don't really like this. I think I should use a hint instead of relying on having the values declared for creating a select element.

Maybe the hint should be something like 'multiple-choice'.